### ISSUE TITLE
Provide kotlin API wrapper

### ISSUE THREAD

**Issue by SokoMishaLov** [lettuce-io/lettuce-core#1386]
It would be great to have Kotlin API wrapper.
For now use of lettuce is good with kotlin and kotlin coroutines:
```kt
suspend fun main() {
    val conn: StatefulRedisConnection<String, String> = TODO()

    val foo1 = conn
            .async()
            .set("foo", "bar")
            .await()

    val foo2 = conn
            .reactive()
            .keys("fo*")
            .collectList()
            .awaitFirstOrNull()
            .orEmpty()
}
```

But I think it could be better.
I propose:

1) Introduce (generate) kotlin suspendable API like others (reactive, async, sync) and generate implementations for
it based on reactive and async APIs.

```kt 
suspend fun main() {
    val cmd: RedisSuspendableCommands<String, String> = TODO()

    val foo1 = cmd.set("foo", "bar")
    val foo2 = cmd.keys("fo*")
}
```

2) Introduce extensions for existing APIs.
For example, I think multiple executed commands could be wrapped with DSL syntax:
```kt
suspend fun main() {
    val conn: StatefulRedisConnection<String, String> = TODO()

    val result: TransactionResult? = conn.async().multi {
        set("foo", "bar")
        get("foo")
    }
}
```

I can't come up with other ideas, but maybe someone could propose something good in the comments below.

Related PR: https://github.com/lettuce-io/lettuce-core/pull/1387

**Comment by mp911de** [comment]
We should be able to generate an API that defines suspended functions and that uses the reactive one underneath. I'm worried a bit regarding nullability as the existing Redis API behaves different when using transactions. Essentially, all API calls return null when using a transaction. 

From the usage example, transactions look pretty decent so I'm torn on transactions.

**Comment by SokoMishaLov** [comment]
That's why a generated API should have nullable return types. Another reason is just to keep safe interop with java APIs.

**Comment by mp911de** [comment]
That fixed with #1387.
